rules/models/rollup.go

```diff
@@ -0,0 +1,229 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package models
+
+import (
+	"sort"
+
+	"github.com/m3db/m3metrics/policy"
+)
+
+// RollupTarget is a common json serializable rollup target.
+type RollupTarget struct {
+	Name     string          `json:"name" validate:"required"`
+	Tags     []string        `json:"tags" validate:"required"`
+	Policies []policy.Policy `json:"policies" validate:"required"`
+}
+
+// RollupRule is a common json serializable rollup rule. Implements Sort interface.
+type RollupRule struct {
+	ID                  string         `json:"id,omitempty"`
+	Name                string         `json:"name" validate:"required"`
+	Filter              string         `json:"filter" validate:"required"`
+	Targets             []RollupTarget `json:"targets" validate:"required,dive,required"`
+	CutoverMillis       int64          `json:"cutoverMillis,omitempty"`
+	LastUpdatedBy       string         `json:"lastUpdatedBy"`
+	LastUpdatedAtMillis int64          `json:"lastUpdatedAtMillis"`
+}
+
+// RollupTargetView is a human friendly representation of a rollup rule target at a given point in time.
```

Comment:
nit: move this down to where `RollupTargetView` methods are.

Reply:
For sure. I generally come from the school of putting all the public types at the top of the file, but that may not be the convention of the repo.